cpp/CMakeLists.txt

```diff
@@ -47,6 +47,13 @@ if(BUILD_WITH_GLPK)
     message(STATUS "Found glpk ${GLPK_LIBRARY} ${GLPK_INCLUDE_DIR}")
 endif(BUILD_WITH_GLPK)
 
+find_package(OpenMP REQUIRED)
```

Comment:
It's better to have a BUILD_WITH_OPENMP flag and find OpenMP only if it is switched on. Otherwise this breaks compilation.

Reply:
OK.

Reply:
And here is actually the implementation of parallelism.